packages/thirdweb/src/wallets/in-app/native/native-connector.ts

```diff
@@ -89,18 +91,37 @@ export class InAppNativeConnector implements InAppConnector {
     }
   }
 
-  async authenticate(params: AuthArgsType): Promise<AuthLoginReturnType> {
+  async authenticate(
+    params: AuthArgsType,
+  ): Promise<AuthStoredTokenWithCookieReturnType> {
+    const strategy = params.strategy;
+    switch (strategy) {
+      case "google":
+      case "facebook":
+      case "discord":
+      case "apple": {
+        const ExpoLinking = require("expo-linking");
+        const redirectUrl =
+          params.redirectUrl || (ExpoLinking.createURL("") as string);
+        return authenticate({ strategy, redirectUrl }, this.options.client);
+      }
+      default:
+        throw new Error(`Unsupported authentication type: ${strategy}`);
+    }
+  }
+
+  async connect(params: AuthArgsType): Promise<AuthLoginReturnType> {
     const strategy = params.strategy;
     switch (strategy) {
       case "email": {
-        return await this.validateEmailOTP({
+        return await this.validateOtp({
           email: params.email,
           otp: params.verificationCode,
           recoveryCode: params.verificationCode,
         });
       }
       case "phone": {
-        return await this.validateEmailOTP({
+        return await this.validateOtp({
```

Comment:
I also always pass my phone number to the email field [![???](https://graphite-user-uploaded-assets-prod.s3.amazonaws.com/default-memes/questionmarks.gif)](https://app.graphite.dev//settings/meme-library?org=thirdweb-dev)

Reply:
Hey man, this whole auth module is jank, we're slowly getting rid of it (don't look at the embedded wallet database if you don't like this)